backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs

```diff
@@ -103,18 +125,81 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
         public async Task<IActionResult> DeleteLink(
             [HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route = "assignments/{assignmentId}/links/{linkId}")] HttpRequest req,
             [Table(AssignmentLinksTableName)] CloudTable assignmentLinksTable,
-            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete)
+            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete,
+            [Assignment(AssignmentId = "{assignmentId}")] Assignment assignment,
+            [Platform] PlatformsClient platformsClient,
+            [LtiAdvantage] INrpsClient nrpsClient,
+            [Lti1] Lti1MembershipClient membershipClient)
         {
             if (entityToDelete == null)
+            {
+                _logger.LogInformation("Entity to delete is null");
                 return new OkResult();
+            }
+
+            IActionResult res = await ValidateUser(req, assignment, platformsClient, nrpsClient, membershipClient);
+            if (res.GetType() != typeof(OkResult))
+                return res;
 
             TableOperation deleteTable = TableOperation.Delete(entityToDelete);
             TableResult deleteResult = await assignmentLinksTable.ExecuteAsync(deleteTable);
 
             if (deleteResult.HttpStatusCode < 200 || deleteResult.HttpStatusCode >= 300)
                 return new InternalServerErrorResult();
 
+            _logger.LogInformation("Link deleted.");
+
+            return new OkResult();
+        }
+
+        private async Task<IActionResult> ValidateUser(HttpRequest req, Assignment assignment, PlatformsClient platformsClient, INrpsClient nrpsClient, Lti1MembershipClient membershipClient)
+        {
+            if (!req.Headers.TryGetUserEmails(out List<string> userEmails))
+            {
+                _logger.LogError("Could not get user email.");
+                return new BadRequestErrorMessageResult("Bad Request: Could not get user email.");

```

Comment:
I don't believe we'd need to add Bad Request in the Message. That should be indicative from `http400` automatically. In case there's a deeper meaning to this, please let me know.

Reply:
No deeper meaning to it, that was added in the flow while writing the code.